src/OrleansSQLUtils/Storage/RelationalOrleansQueries.cs

```diff
@@ -833,13 +833,20 @@ private static ReminderEntry CreateReminderEntry(IDataRecord record)
 
         private static IDbDataParameter CreateBeginHashParameter(IDbCommand command, uint beginHash)
         {            
-            return command.CreateParameter(ParameterDirection.Input, "beginHash", (int)beginHash);
+            return CreateHashRangeParameter(command, "beginHash", beginHash);
         }
 
 
         private static IDbDataParameter CreateEndHashParameter(IDbCommand command, uint endHash)
         {            
-            return command.CreateParameter(ParameterDirection.Input, "endHash", (int)endHash);
+            return CreateHashRangeParameter(command, "endHash", endHash);
+        }
+
+
+        private static IDbDataParameter CreateHashRangeParameter(IDbCommand command, string paramName, uint hash)
+        {
+            const uint delta = (uint) int.MaxValue + 1;
+            return command.CreateParameter(ParameterDirection.Input, paramName, (int)(hash - delta));
```

Comment:
Since this line relies on unchecked arithmetic, I wonder if it should be explicitly stated with `unchecked`? The compiler switch defaults to this (`/checked = false`), but maybe it conveys information of the intention and guards against inadvertent compiler switch change.